llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1076,6 +1078,11 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
 
   ModulePassManager MPM;
 
+  if (Phase != ThinOrFullLTOPhase::ThinLTOPostLink) {
```

Comment:
You'll also want to do this at O0, where this function isn't called, for the supported case of ThinLTO + O0.